### PR TITLE
DOCSP-42313 Connection Targets

### PR DESCRIPTION
**Pull request by lindseymoore**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-scala/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-42313>
Staging - <https://deploy-preview-80--docs-scala.netlify.app/connect/connection-targets/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/includes/connect/connection-target.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+// start-connect
+object MongoClientConnectToAtlas {
+
+  def main(args: Array[String]): Unit = {
+
+    // Replace the placeholder with your Atlas connection string
+    val connectionString = "<connection string>";
+
+    // Construct a ServerApi instance using the ServerApi.builder() method
```
Comment (stephmarie17) [inline source/includes/connect/connection-target.scala:17]: [s] Can update the narrative code comments to use indicative mood per the code comment guide [here](https://docs.google.com/document/d/1Bnhh1OsGRUuXo_6yVUASGWiSX6sWuqrEFxPP6sLZDHM/edit?tab=t.0#heading=h.xv6glj2lnzjh). For example:
```suggestion
    // Constructs a ServerApi instance using the ServerApi.builder() method
```

File: source/connect/connection-targets.txt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+.. note::
+
+   The ``MongoClient`` constructor is *non-blocking*.
+   When you connect to a replica set, the constructor returns immediately while the
+   client uses background threads to connect to the replica set.
+   
+   If you construct a ``MongoClient`` and immediately print the string representation
+   of its ``nodes`` attribute, the list might be empty while the client connects to
+   the replica-set members.
```
Comment (stephmarie17) [inline source/connect/connection-targets.txt:139]: [s] Can potentially link to any relevant API documentation here? It looks like other drivers are linking to `MongoClient` documentation but not sure if that applies for Scala or not 😅

File: source/connect/stable-api.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: compatible, backwards, upgrade
```
Comment (stephmarie17) [inline source/connect/stable-api.txt:18]: [s] Might be worth updating the keywords now so it doesn't get missed when this page is written.

File: source/connect.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.. _scala-connect:
```
Comment (stephmarie17) [inline source/connect.txt:1]: [q] Should this page be accessed when selecting `Connect to MongoDB` in the ToC? It didn't show up for me but I see it if I manually enter the URL. Not sure if that is part of this PR, but I think it just needs to be added to snooty.toml.

File: source/index.txt [hunk 5]
```diff
@@ -13,6 +13,7 @@
    :maxdepth: 1
```
Comment (stephmarie17) [inline source/index.txt:13]: [q] Not part of this diff, but is there a section here to comment in that links to the new Connection pages?

**Review by stephmarie17 — CHANGES_REQUESTED** [review]
Nice work! Content looks good but just had some questions about set-up and linking. I know this project is a work in progress, but wanted to verify before approving 😄

File: source/connect/stable-api.txt (hunk 3 above)
Comment (lindseymoore) [reply source/connect/stable-api.txt:18]: Stable API page already completed!

File: source/connect.txt (hunk 4 above)
Comment (lindseymoore) [reply source/connect.txt:1]: No ticket to add a landing page for this Connect section in the epic. Will most likely be added in a future piece of work!

File: source/index.txt (hunk 5 above)
Comment (lindseymoore) [reply source/index.txt:13]: The `\connect` entry links to connect.txt. The links for the pages in the Connect section can be found in the connect.txt file: https://github.com/mongodb/docs-scala/pull/80/files#diff-b097e010c9e7918edece06cd4f7dc5e362cd66f61aa38035127990c2c4dac66e.

**Review by stephmarie17 — APPROVED** [review]
LGTM 🚢

File: source/connect/connection-targets.txt [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+- Explicitly specify the ``hostname`` and the ``port``:
+
+  .. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-connect-local-host-port
+   :end-before: end-connect-local-host-port
+   :dedent:
+
+Replica Sets
```
Comment (jyemin) [inline source/connect/connection-targets.txt:88]: Have we decided in general not to show a sharded cluster example?

**Review by jyemin — APPROVED** [review]
One question, but LGTM if the answer is that we're not showing sharded cluster example.

File: source/connect/connection-targets.txt (hunk 6 above)
Comment (lindseymoore) [reply source/connect/connection-targets.txt:88]: @jyemin The team hasn't included information on connecting to a sharded cluster in any of the other drivers, but we can look into whether this makes sense. Thanks for the suggestion!

File: source/connect.txt [hunk 7]
```diff
@@ -13,20 +13,20 @@ Connect to MongoDB
...
    :description: Learn how to use the Scala driver to connect to MongoDB.
    :keywords: client, ssl, tls, localhost
 
 .. toctree::
-    
+
    /connect/mongoclient
    /connect/stable-api
+   /connect/connection-targets
+   /connect/connection-options
```
Comment (lindseymoore) [inline source/connect.txt:26]:
```suggestion
..   /connect/connection-options
```

File: source/connect.txt [hunk 8]
```diff
@@ -13,20 +13,20 @@ Connect to MongoDB
...
 .. toctree::
-    
+
    /connect/mongoclient
    /connect/stable-api
-   /connect/tls
-..   /connect/connection-targets
+   /connect/connection-targets
 ..   /connect/connection-options
+   /connect/tls
```
Comment (lindseymoore) [inline source/connect.txt:27]:
```suggestion
   /connect/connection-targets
   /connect/tls
..   /connect/connection-options
```

**Pull request merged**